packages/components/utils/src/useKeyboard/useKeyboard.ts

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export interface UseKeyboardProps {
+  /**
+   * @description defines which element is the event attached to
+   * @default document
+   */
+  ref?: Document | Window | HTMLElement;
+  /**
+   * @description defines which key to look for i.e. `ArrowUp`, `Escape`, `Shift`
+   */
+  key: KeyboardEvent['key'];
```

Comment:
This resolves to a `string`, I couldn't find the proper types with the full list of keys, any ideas?

Reply:
I don't think such a type exists so this is probably fine.